src/components/application-bar/ApplicationBar.js

```diff
@@ -38,12 +38,12 @@ function ApplicationBar({
               {drawerMenu}
             </DrawerMenu>
           </div>
-          <Typography variant="h6" color="inherit" className={classes.grow} noWrap>
+          <Typography variant="h6" color="inherit" data-test={title} className={classes.grow} noWrap>
             {title}
-            {build && <Typography variant="caption" color="inherit" > build {build}</Typography>}
+            {build && <Typography variant="caption" color="inherit" data-test={build} > build {build}</Typography>}
           </Typography>
-          <Typography variant="subtitle2" color="inherit" className={classes.grow} noWrap>
-            {filepath || ''}
```

Comment:
filepath and file.filepath are two different things. If a file exists, it gets the file path from it. If the filepath is passed from the calling component, it can be used. So maybe we do a combo?